src/PioneerUtils.jsm

```diff
@@ -161,7 +167,52 @@ export class PioneerUtils {
     const hashKey = `${this.config.studyName}/${await this.getPioneerId()}`;
     return sampling.chooseWeighted(this.config.branches, hashKey);
   }
+
+  /**
+   * Ends a study by uninstalling the addon and sending a relevant
+   * event ping to telemetry.
+   *
+   * @param {String?} eventId
+   *   The ID of the event that occured.
+   * 
+   * @returns {String}
+   *   The ID of the event ping that was submitted.
+   */
+  endStudy(eventId) {
+    if (eventId === undefined) {
+      eventId = PioneerUtils.EVENTS.ENDED_NEUTRAL;
+    } else if (!(event in PioneerUtils.EVENTS)) {
+      throw new Error("Invalid event ID.");
+    }
+    this.uninstall();
+    return this.submitEncryptedPing("event", 1, { eventId });
```

Comment:
I imagine that most of the event stuff here should be another method, since it seems pretty generic and we'll want to use it in other places as well.